modules/backend/clinic/Clinic.hpp

```diff
@@ -1,23 +1,28 @@
 #pragma once
 
-#include "modules/backend/clinic/staff/Staff.hpp"
-#include "modules/backend/clinic/staff/room/iRoom.hpp"
 #include <memory>
 #include <vector>
 
 namespace clinic
 {
 
+namespace staff
+{
+class Staff;
+}
+
+namespace room
+{
+class iRoom;
+}
```

Comment:
Why are these necessary? 
Interfaces headers are not heavy, they can be included here, in the header. This is the first time I am seeing such thing tbh

Reply:
Forward declarations here are necessary after I moved includes to .cpp files.

I try to stick to the recommendation to include only what is necessary and optimally - in cpp files only.
It is hard to say which is better, as conflicting opinions exist.
In some cases it could speedup compilation. Hard to observe now but maybe as the project grows...

If we decide to resign from forward declarations in favor of includes no great harm will come in our case I think.
But it should be deliberate decision, which we might include in our guidelines.
Some little research for pros and cons might be good beforehand.